src/com/koxudaxi/poetry/PyAddExistingPoetryEnvPanel.kt

```diff
@@ -55,20 +58,32 @@ class PyAddExistingPoetryEnvPanel(private val project: Project?,
                 .panel
         add(formPanel, BorderLayout.NORTH)
         addInterpretersAsync(sdkComboBox) {
-            detectPoetryEnvs(module, existingSdks, context, project?.basePath ?: newProjectPath)
+            val existingSdkPaths = sdkHomes(existingSdks)
+            val moduleSdks = allModules(project).parallelStream().flatMap { module ->
```

Comment:
Does not take a crazy long time to load if there are many modules (I have 45):
<img width="866" alt="Add_Python_Interpreter_and_Project_Structure" src="https://user-images.githubusercontent.com/527952/116504964-9e9be400-a887-11eb-8581-764001121d30.png">